src/components/app/AssemblyViewToggleButton.tsx

```diff
@@ -37,7 +39,11 @@ class AssemblyViewToggleButtonImpl extends React.PureComponent<Props> {
   };
 
   override render() {
-    const { assemblyViewIsOpen } = this.props;
+    const { assemblyViewIsOpen, isAssemblyViewAvailable } = this.props;
+
+    if (!isAssemblyViewAvailable) {
+      return null;
+    }
```

Comment:
I disagree with this change; if we add more buttons I don't want them to shift position depending on context. Is the disabled state not disabled enough?